steps/container.py

```diff
@@ -233,3 +185,71 @@ def get_output(self, history=True):
     def remove_image(self, force=False):
         self.logging.info("Removing image %s" % self.image_id)
         d.remove_image(image=self.image_id, force=force)
+
+    def copy_file_to_container(self, src_file, dest_folder):
+        full_src_file=os.path.join(os.getcwd(), src_file)
+
+        # The Docker library needs tar bytes to put_archive
+        f = tempfile.NamedTemporaryFile()
+        t = tarfile.open(mode='w', fileobj=f)
+        abs_path = os.path.abspath(full_src_file)
+        t.add(abs_path, arcname=os.path.basename(full_src_file), recursive=False)
+        t.close()
+        f.seek(0)
+        try:
+            d.put_archive(
+                container=self.container['Id'],
+                path=dest_folder,
+                data=f.read())
```

Comment:
Here you could have another `with` statement, this time to only read the content.